bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplacer.java

```diff
@@ -0,0 +1,790 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.ui.texteditor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.PatternSyntaxException;
+
+import org.eclipse.swt.custom.BusyIndicator;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.widgets.Display;
+
+import org.eclipse.jface.text.IFindReplaceTarget;
+import org.eclipse.jface.text.IFindReplaceTargetExtension;
+import org.eclipse.jface.text.IFindReplaceTargetExtension3;
+import org.eclipse.jface.text.IFindReplaceTargetExtension4;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.Region;
+
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.internal.texteditor.NLSUtility;
+
+/**
+ * @since 3.17
+ */
+public class FindReplacer {
```

Comment:
I am not convinced that `FindReplacer` is a good name. It emphasizes "replacer" although it does not necessarily replace something. I am not sure what a good name would be, but I would at least prefer something like a simply `FindAndReplace`.

Reply:
I find inventing a good name for this class to be hard, since it is abstract by nature. I will go for your suggestion of `FindAndReplace` for now (and I agree that this is a really good name, far better than `FindReplacer`), but want to document a few more ideas that we may choose to use in subsequent review cycles:
- `FindReplaceService` https://stackoverflow.com/questions/1860852/business-logic-classes-naming
- `FindReplaceLogic` https://softwareengineering.stackexchange.com/questions/259838/naming-conventions-dal-bal-and-ui-layer